src/OutputEllipseComponent.cpp

```diff
@@ -38,9 +53,152 @@ void OutputEllipseComponent::paint(Graphics &g)
+					g.setColour(Colour::fromFloatRGBA(lineColour.r, lineColour.g, lineColour.b, 1.0f));
+					g.drawEllipse(line->get_width() / 2.0f, line->get_width() / 2.0f, get_width() - line->get_width(), get_height() - line->get_width(), line->get_width());
+				}
+			}
+			else
+			{
+				// Juce coordinate system 0° is at the Y axis positive, calculating clockwise
+				// IsoBus coordinate system 0° is at the X axis positive, calculating counter-clockwise
+				float startAngle = juce::degreesToRadians(((get_start_angle() * 2.0f)));
+				float endAngle = juce::degreesToRadians(((get_end_angle() * 2.0f)));
+
+				juce::Path arcPath;
+				if (get_ellipse_type() == isobus::OutputEllipse::EllipseType::ClosedEllipseSegment)
+				{
+					// segment: the ellipse section endpoints connected to the center with two lines
+					arcPath.startNewSubPath(centerX, centerY);
+				}
+
+				float wOffset = line->get_width() / 2.0f;
+
+				addArcToPath(arcPath, wOffset, wOffset, get_width() - line->get_width(), get_height() - line->get_width(), startAngle, endAngle, get_ellipse_type() != isobus::OutputEllipse::EllipseType::ClosedEllipseSegment);
+
+				if (get_ellipse_type() == isobus::OutputEllipse::EllipseType::ClosedEllipseSegment)
+				{
+					// segment: the ellipse section endpoints connected to the center with two lines
+					arcPath.lineTo(centerX, centerY);
+					if (fillNeeded)
+					{
+						g.setColour(Colour::fromFloatRGBA(fillColour.r, fillColour.g, fillColour.b, 1.0f));
+						g.fillPath(arcPath);
+					}
+
+					g.setColour(Colour::fromFloatRGBA(lineColour.r, lineColour.g, lineColour.b, 1.0f));
+					g.strokePath(arcPath, juce::PathStrokeType(line->get_width()));
+				}
+				else if (get_ellipse_type() == isobus::OutputEllipse::EllipseType::ClosedEllipseSection)
+				{
+					// section: the ellipse section endpoints connected with a straight line
+					arcPath.closeSubPath();
+					if (fillNeeded)
+					{
+						g.setColour(Colour::fromFloatRGBA(fillColour.r, fillColour.g, fillColour.b, 1.0f));
+						g.fillPath(arcPath);
+					}
+
+					g.setColour(Colour::fromFloatRGBA(lineColour.r, lineColour.g, lineColour.b, 1.0f));
+					g.strokePath(arcPath, juce::PathStrokeType(line->get_width()));
+				}
+				else if (get_ellipse_type() == isobus::OutputEllipse::EllipseType::OpenDefinedByStartEndAngles)
+				{
+					g.setColour(Colour::fromFloatRGBA(lineColour.r, lineColour.g, lineColour.b, 1.0f));
+					g.strokePath(arcPath, juce::PathStrokeType(line->get_width()));
+				}
+			}
+		}
+	}
+}
+
+/**
+ * @brief OutputEllipseComponent::addArcToPath
+ * Method to draw ellipse segments to keep the angle between the start and end accurate.
+ * Juce Path::addArc does not keep the angle when drawing.
+ */
+void OutputEllipseComponent::addArcToPath(Path &path, float x, float y, float w, float h, float fromRadians, float toRadians, bool startAsNewSubPath) const
+{
+	float a = w / 2;
+	float b = h / 2;
```

Comment:
```suggestion
	float a = w / 2.0f;
	float b = h / 2.0f;
```
`a` and `b` aren't very descriptive names